sdk/core/utils/redisclient/redisclient_test.go

```
@@ -0,0 +1,645 @@
+	}
+}
+
+// TestGetOrCreateClient_DoesNotHoldLockDuringPing proves the registry lock
+// guards only the map lookup/insert, never c.Ping - mu is process-wide, so
+// holding it during a slow/unreachable Redis's ping would stall every other
+// caller too, even for an unrelated, healthy endpoint.
+func TestGetOrCreateClient_DoesNotHoldLockDuringPing(t *testing.T) {
+	// Accepts but never responds, so Ping against it blocks until the
+	// deadline - a reliable window to prove a concurrent, unrelated key
+	// isn't blocked by it.
+	ln, err := net.Listen("tcp", "127.0.0.1:0")
```

golangci-lint (2.12.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)